src/lib/markdown-report-builder.ts

```diff
@@ -11,18 +11,37 @@ export default class MarkdownReportBuilder {
     return multiline(`
       # Extension Updates
 
+      ${this.buildToc(extensions)}
       ${this.buildExtension(extensions)}
       `);
   }
 
+  private buildToc(extensions: LoadedExtension[]): string {
+    if (extensions.length < 2) return '';
+    return multiline(`
+      <details>
+        <summary>There are ${extensions.length} extensions updated.</summary>
+
+        ${this.buildTocLink(extensions)}
+      </details>
+      `);
+  }
+
+  private buildTocLink(extensions: LoadedExtension[]): string {
+    return extensions
+      .map((extension: LoadedExtension, index: number): string =>
+        `${index+1}. [${extension.displayName}](#${extension.id})`
+      )
+      .join('\n');
+  }
+
   private buildExtension(extensions: LoadedExtension[]) {
     return extensions
       .map(extension =>
         multiline(`
         <details open>
           <summary>
-
-          ## ${extension.displayName} \`${extension.id}\`
+            <h2 id="${extension.id}">${extension.displayName} <code>${extension.id}</code></h2>
```

Comment:
Wonder if we could add an extension ID as an anchor with markdown notation. 

Looking at [this stackoverflow question](https://stackoverflow.com/questions/5319754/cross-reference-named-anchor-in-markdown), if `{#my-anchor}` (or its variation) works, it would be great. Otherwise `<a name="my-anchor"></a>` looks the second best.

So that we don't need to think about escaping HTML characters in `displayName` ourselves, like the case `</h2><a href="http://somewhere-doggy">Extension name</a><h2>`

Reply:
I'll investigate the suggested solutions, but I don't think those will work, as most are custom implementation for GitBook, rmarkdown, Gruber Markdown, pandoc, Hugo, or any other service/parser.

The `<a name="my-anchor"></a>` is an good alternative. I'll try it.

> [..] like the case `</h2><a href="http://somewhere-doggy">Extension name</a><h2>`

I highly doubt Microsoft will allow those names as an extension, but I get your point. 👍

Reply:
Thanks for looking into this!! 🙏 Great to hear that `<a id=` worked for us! 🎉